OpenRA.Game/Map/TileSet.cs

```diff
@@ -246,6 +246,9 @@ public byte GetTerrainIndex(string type)
 
 		public byte GetTerrainIndex(TerrainTile r)
 		{
+			if (!Templates.ContainsKey(r.Type))
```

Comment:
You may also need 965447a42c92aecdbeacef69be4f72a1478ef08b for the removed tiles to not crash in certain other cases.